### PR TITLE
Add multi-lingual entities

### LINKED ISSUE #1754: Multi lingual entities
Hello!

[Multi lingual entities](https://github.com/aspnetboilerplate/aspnetboilerplate/tree/dev/src/Abp/Domain/Entities) were realized recently in ASP.NET Boilerplate [Entites](https://aspnetboilerplate.com/Pages/Documents/Multi-Lingual-Entities).

It supports IMultiLingualEntity, IEntityTranslation, Translations, MultiLingualMap, MultiLingual CRUD operation.

Please, realize, multi-lingual entities in Abp.Io [Entites](https://github.com/abpframework/abp/tree/dev/framework/src/Volo.Abp.Ddd.Domain/Volo/Abp/Domain/Entities).

Regards, Sergey

### PR DESCRIPTION
**Pull request by realLiangshiwei**
Resolve https://github.com/abpframework/abp/issues/1754

### CODE REVIEW

File: framework/test/Volo.Abp.AutoMapper.Tests/AutoMapper/AbpAutoMapperMultiLingualDtoExtensions_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                        Name = "深入理解C#"
+                    }
+                }
+            };
+        }
+
+        [Fact]
+        public void Should_Map_CurrentUICulture()
+        {
+            CultureInfo.CurrentCulture = new CultureInfo("en");
```
Comment (maliming) [inline framework/test/Volo.Abp.AutoMapper.Tests/AutoMapper/AbpAutoMapperMultiLingualDtoExtensions_Tests.cs]: `using (CultureHelper.Use("en"))`

File: framework/src/Volo.Abp.Ddd.Domain/Volo/Abp/Domain/Entities/IEntityTranslation.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public interface IEntityTranslation<TEntity, TPrimaryKeyOfMultiLingualEntity> : IEntityTranslation
+        where TEntity : IEntity<TPrimaryKeyOfMultiLingualEntity>
+    {
+        TEntity Core { get; set; }
+
+        TPrimaryKeyOfMultiLingualEntity CoreId { get; set; }
+    }
+
+    public interface IEntityTranslation<TEntity>: IEntityTranslation<TEntity, int>
+        where TEntity : IEntity<int>
```
Comment (maliming) [inline framework/src/Volo.Abp.Ddd.Domain/Volo/Abp/Domain/Entities/IEntityTranslation.cs]: If this interface is necessary then use `Guid`.

File: framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/Modeling/AbpEntityTypeBuilderExtensions.cs [hunk 3]
```diff
@@ -302,6 +304,25 @@ public static void ConfigureFullAuditedAggregateRoot<T>(this EntityTypeBuilder<T
...
+        {
+            b.As<EntityTypeBuilder>().TryConfigureEntityTranslation();
+        }
+
+        public static void TryConfigureEntityTranslation(this EntityTypeBuilder b)
+        {
+            if (b.Metadata.ClrType.IsAssignableTo<IMultiLingualTranslation>())
+            {
+                b.HasIndex(nameof(IMultiLingualTranslation.Language))
+                    .IsUnique();
```
Comment (hikalkan) [inline framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/Modeling/AbpEntityTypeBuilderExtensions.cs]: What about multi-tenancy? Making this unique prevents multiple tenants to insert entity with the same language name.

Comment (enisn) [reply framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/Modeling/AbpEntityTypeBuilderExtensions.cs]: Combination of Tenant & Language can be unique @realLiangshiwei

**Review by maliming — APPROVED** [review]

**Pull request merged**